agent/xds/clusters.go

```diff
@@ -956,6 +958,7 @@ func (s *ResourceGenerator) makeUpstreamClusterForPreparedQuery(upstream structs
 }
 
 func (s *ResourceGenerator) makeUpstreamClustersForDiscoveryChain(
+	listenerKey proxycfg.IngressListenerKey,
```

Comment:
It feels weird having this as a param to a function that gets called in non-ingress-gw contexts.

Could injecting/overriding the max connections happen in `clustersFromSnapshotIngressGateway`? Not sure what's a cleaner solution

Reply:
Done